kornia/color/colormap.py

```diff
@@ -1,145 +1,182 @@
-) -> Tensor:
-    return tensor(list(colors), dtype=dtype, device=device).T
+    List of available colormaps:
 
+    .. image:: _static/img/CMAP.png
+    """
 
-class ColorMap(ABC):
-    r"""Abstract class to represents a color map."""
+    autumn = 1
+    bone = 2
+    jet = 3
+    winter = 4
+    rainbow = 5
+    ocean = 6
+    summer = 7
+    spring = 8
+    cool = 9
+    hsv = 10
+    brg = 11
+    pink = 12
+    hot = 13
+    plasma = 14
+    viridis = 15
+    cividis = 16
+    twilight = 17
+    turbo = 18
+    seismic = 19
+
+    def _load_base(self) -> list[RGBColor]:
+        r"""Load the base colormap corresponding to the enumeration member.
+
+        Returns:
+            The base colormap.
+        """
+        return {
+            "autumn": cm_data.get_autumn_base,
+            "bone": cm_data.get_bone_base,
+            "jet": cm_data.get_jet_base,
+            "winter": cm_data.get_winter_base,
+            "rainbow": cm_data.get_rainbow_base,
+            "ocean": cm_data.get_ocean_base,
+            "summer": cm_data.get_summer_base,
+            "spring": cm_data.get_spring_base,
+            "cool": cm_data.get_cool_base,
+            "hsv": cm_data.get_hsv_base,
+            "brg": cm_data.get_bgr_base,
+            "pink": cm_data.get_pink_base,
+            "hot": cm_data.get_hot_base,
+            "plasma": cm_data.get_plasma_base,
+            "viridis": cm_data.get_viridis_base,
+            "cividis": cm_data.get_cividis_base,
+            "twilight": cm_data.get_twilight_base,
+            "turbo": cm_data.get_turbo_base,
+            "seismic": cm_data.get_seismic_base,
+        }[self.name]()
+
+    @classmethod
+    def list(cls) -> list[str]:
+        r"""Returns a list of names of enumeration members.
+
+        Returns:
+            A list containing the names of enumeration members.
+        """
+        return [c.name for c in cls]
+
+
+class ColorMap:
+    r"""Class to represent a colour map. It can be created or selected from the built-in colour map. Please refer to
+    the `CMAP` enum class to view all available colormaps.
 
-    colors: Tensor
+    Args:
+        base: A list of RGB colors to define a new custom colormap or
+        the name of a built-in colormap as str or using CMAP class.
+        num_colors: Number of colors in the colormap.
+        device: The device to put the generated colormap on.
+        dtype: The data type of the generated colormap.
+
+    Returns:
+        An object of the colormap with the num_colors length.
+
+    Examples:
+        >>> ColorMap(base='viridis', num_colors=8).colors
+        tensor([[0.2813, 0.2621, 0.2013, 0.1505, 0.1210, 0.2463, 0.5259, 0.8557],
+                [0.0842, 0.2422, 0.3836, 0.5044, 0.6258, 0.7389, 0.8334, 0.8886],
+                [0.4072, 0.5207, 0.5543, 0.5574, 0.5334, 0.4519, 0.2880, 0.0989]])
+
+        >>> # Create color map from first color (RGB with range[0-1]) to last one with num_colors length.
```

Comment:
This will be better displayed 
```suggestion
    Create a color map from the first color (RGB with range[0-1]) to the last one with num_colors length.
```